k2/csrc/ragged_test.cu

```diff
@@ -1430,53 +1391,148 @@ TEST(RaggedShapeOpsTest, TestRenumber) {
+    Ragged<int32_t> ragged(shape, col_indexes);
+    Array1<int32_t> order = GetTransposeReordering(ragged, 6);
+    CheckArrayData(order, {2, 5, 0, 4, 6, 3, 1, 7, 9, 8, 10});
     EXPECT_TRUE(context->IsCompatible(*order.Context()));
   }
 }
 
 TEST(GetTransposeReordering, WithDuplicates) {
-  // 0 0 0 (9,9,9)
-  // 5 8 0     0
-  // 0 0 (3,3) 0
-  // 0 6 0     0
-  std::vector<int32_t> col_indexes{3, 3, 3, 0, 1, 2, 2, 1};
-  std::vector<int32_t> _row_splits{0, 3, 5, 7, 8};
+  //       col0   col1   col2    col3      col4      col5
+  // row0         a0,a1         b2,b3,b4
+  // row1  c5,c6          d7
+  // row2         e8
+  // row3   f9   g10,g11         h12
+  // row4                                i13,i14,i15
+  // row5                        j16                  k17
+  std::vector<int32_t> col_indexes{1, 1, 3, 3, 3, 0, 0, 2, 1,
+                                   0, 1, 1, 3, 4, 4, 4, 3, 5};
+  std::vector<int32_t> _row_splits{0, 5, 8, 9, 13, 16, 18};
   for (auto &context : {GetCpuContext(), GetCudaContext()}) {
     Array1<int32_t> row_splits(context, _row_splits);
     RaggedShape shape = RaggedShape2(&row_splits, nullptr, -1);
     Array1<int32_t> values(context, col_indexes);
-
     Ragged<int32_t> ragged(shape, values);
-    Array1<int32_t> order = GetTransposeReordering(ragged, 4);
-    //   index 0 1 2 3 4 5 6 7
-    // it maps 9 9 9 5 8 3 3 6 to
-    //         5 8 6 3 3 9 9 9
-    // so it returns
-    //         3 4 7 5 6 0 1 2   Note that it is stable
-    CheckArrayData(order, {3, 4, 7, 5, 6, 0, 1, 2});
+    Array1<int32_t> order = GetTransposeReordering(ragged, 6);
+    CheckArrayData(
+        order, {5, 6, 9, 0, 1, 8, 10, 11, 7, 2, 3, 4, 12, 16, 13, 14, 15, 17});
+    EXPECT_TRUE(context->IsCompatible(*order.Context()));
+  }
+}
+
+TEST(GetTransposeReordering, WithDuplicatesThreeAxes) {
+  //       col0   col1   col2    col3      col4      col5
+  // row0         a0,a1         b2,b3,b4
+  // row1  c5,c6          d7
+  // row2         e8
+  // row3   f9   g10,g11         h12
+  // row4                                i13,i14,i15
+  // row5                                 j16         k17
+  for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+    Array1<int32_t> col_indexes(
+        context, std::vector<int32_t>{1, 1, 3, 3, 3, 0, 0, 2, 1, 0, 1, 1, 3, 4,
+                                      4, 4, 4, 5});
+    Array1<int32_t> row_splits1(context, std::vector<int32_t>{0, 4, 6});
+    Array1<int32_t> row_splits2(context,
+                                std::vector<int32_t>{0, 5, 8, 9, 13, 16, 18});
+    RaggedShape shape =
+        RaggedShape3(&row_splits1, nullptr, -1, &row_splits2, nullptr, -1);
+    Ragged<int32_t> ragged(shape, col_indexes);
+    Array1<int32_t> order = GetTransposeReordering(ragged, 6);
+    CheckArrayData(
+        order, {5, 6, 9, 0, 1, 8, 10, 11, 7, 2, 3, 4, 12, 13, 14, 15, 16, 17});
     EXPECT_TRUE(context->IsCompatible(*order.Context()));
   }
 }
 
+TEST(ChangeSublistSize, TwoAxes) {
```

Comment:
And I think we can test `ChangeSublistSize` with a random size, e.g. `RandomRaggedShape(false, 2,4,0,1000)` and `size_delta = RandInt(0, random_shape.MaxSize(NumAxes())'.

Reply:
I'll add tests with random sizes.